external/deep-object-reid/torchreid_tasks/openvino_task.py

```diff
@@ -183,19 +183,35 @@ def infer(self, dataset: DatasetEntity,
-            predicted_scene, actmap, repr_vector, act_score = self.inferencer.predict(dataset_item.numpy)
+            predicted_scene, saliency_map, repr_vector, act_score = self.inferencer.predict(dataset_item.numpy)
             dataset_item.append_labels(predicted_scene.annotations[0].get_labels())
             active_score_media = FloatMetadata(name="active_score", value=act_score,
                                                float_type=FloatType.ACTIVE_SCORE)
             dataset_item.append_metadata_item(active_score_media, model=self.model)
             feature_vec_media = TensorEntity(name="representation_vector", numpy=repr_vector.reshape(-1))
             dataset_item.append_metadata_item(feature_vec_media, model=self.model)
             if dump_features:
-                saliency_media = ResultMediaEntity(name="Saliency Map", type="saliency_map",
-                                                   annotation_scene=dataset_item.annotation_scene,
-                                                   numpy=actmap, roi=dataset_item.roi,
-                                                   label=predicted_scene.annotations[0].get_labels()[0].label)
-                dataset_item.append_metadata_item(saliency_media, model=self.model)
+                if saliency_map.ndim == 2:
+                    # Single saliency map per image, support e.g. EigenCAM use case
+                    actmap = get_actmap(saliency_map, (dataset_item.width, dataset_item.height))
+                    saliency_media = ResultMediaEntity(name="Saliency Map", type="saliency_map",
+                                                       annotation_scene=dataset_item.annotation_scene,
+                                                       numpy=actmap, roi=dataset_item.roi,
+                                                       label=predicted_scene.annotations[0].get_labels()[0].label)
+                    dataset_item.append_metadata_item(saliency_media, model=self.model)
+                elif saliency_map.ndim == 3:
+                    # Multiple saliency maps per image (class-wise saliency map), support e.g. Recipro-CAM use case
+                    for class_id, class_wise_saliency_map in enumerate(saliency_map):
+                        actmap = get_actmap(class_wise_saliency_map, (dataset_item.width, dataset_item.height))
+                        class_name_str = self.task_environment.get_labels(include_empty=True)[class_id].name
+                        saliency_media = ResultMediaEntity(name=class_name_str, type="saliency_map",
+                                                           annotation_scene=dataset_item.annotation_scene,
+                                                           numpy=actmap, roi=dataset_item.roi,
+                                                           label=predicted_scene.annotations[0].get_labels()[0].label)
+                        dataset_item.append_metadata_item(saliency_media, model=self.model)
```

Comment:
"name" will be shown in Geti UI in saliency map drop down. It needs to be consistent across tasks. In semantic segmentation, we use same name and different "label".
Could you either follow the convention or fix segmentation side to align? Formats like f"Saliency Map: {class_name_str}" would also be an option.

Reply:
I agree that it needs to be consistent across tasks. Do you mind if I will update all tasks sequentially? Because current PR only update classification.
I personally like `f"Saliency Map: {class_name_str}"` so I used it. I also think that in the cls/det use cases we need to keep the same label for all saliency maps - having true/pseudo label and class-wise saliency maps helps for model debugging.

Reply:
Sounds good. Thank you!